panel/backends/wayland/lxqttaskbarbackendwayland.cpp

```diff
@@ -1,722 +1,234 @@
-    ILXQtTaskbarAbstractBackend(parent)
-{
-    m_managment.reset(new LXQtTaskBarPlasmaWindowManagment);
-    m_workspaceInfo.reset(new LXQtPlasmaWaylandWorkspaceInfo);
-
-    connect(m_managment.get(), &LXQtTaskBarPlasmaWindowManagment::windowCreated, this, [this](LXQtTaskBarPlasmaWindow *window) {
-        connect(window, &LXQtTaskBarPlasmaWindow::initialStateDone, this, [this, window] {
-            addWindow(window);
-        });
-    });
-
-    // connect(m_managment.get(), &LXQtTaskBarPlasmaWindowManagment::stackingOrderChanged,
-    //         this, [this](const QString &order) {
-    //     // stackingOrder = order.split(QLatin1Char(';'));
-    //     // for (const auto &window : std::as_const(windows)) {
-    //     //     this->dataChanged(window.get(), StackingOrder);
-    //     // }
-    // });
-
-    connect(m_workspaceInfo.get(), &LXQtPlasmaWaylandWorkspaceInfo::currentDesktopChanged, this,
-            [this](){
-                int idx = m_workspaceInfo->position(m_workspaceInfo->currentDesktop());
-                idx += 1; // Make 1-based
-                emit currentWorkspaceChanged(idx);
-            });
-
-    connect(m_workspaceInfo.get(), &LXQtPlasmaWaylandWorkspaceInfo::numberOfDesktopsChanged,
-            this, &ILXQtTaskbarAbstractBackend::workspacesCountChanged);
-
-    connect(m_workspaceInfo.get(), &LXQtPlasmaWaylandWorkspaceInfo::desktopNameChanged,
-            this, [this](int idx) {
-        emit workspaceNameChanged(idx + 1); // Make 1-based
-    });
-}
-
-bool LXQtTaskbarWaylandBackend::supportsAction(WId windowId, LXQtTaskBarBackendAction action) const
-{
-    LXQtTaskBarPlasmaWindow *window = getWindow(windowId);
-    if(!window)
-        return false;
-
-    LXQtTaskBarPlasmaWindow::state state;
 
-    switch (action)
-    {
-    case LXQtTaskBarBackendAction::Move:
-        state = LXQtTaskBarPlasmaWindow::state::state_movable;
-        break;
+        else if ( desktop == "wayfire" )
+        {
+            qDebug() << "--------------> Using wayfire backend";
+            m_backend = new LXQtTaskbarWlrootsBackend();
+            break;
+        }
 
-    case LXQtTaskBarBackendAction::Resize:
-        state = LXQtTaskBarPlasmaWindow::state::state_resizable;
-        break;
+        else if ( desktop == "sway" )
+        {
+            qDebug() << "--------------> Using sway backend";
+            m_backend = new LXQtTaskbarWlrootsBackend();
+            break;
+        }
 
-    case LXQtTaskBarBackendAction::Maximize:
-        state = LXQtTaskBarPlasmaWindow::state::state_maximizable;
-        break;
+        else if ( desktop == "labwc" )
+        {
+            qDebug() << "--------------> Using labwc backend";
+            m_backend = new LXQtTaskbarWlrootsBackend();
+            break;
+        }
 
-    case LXQtTaskBarBackendAction::Minimize:
-        state = LXQtTaskBarPlasmaWindow::state::state_minimizable;
-        break;
+        else if ( desktop == "hyprland" )
```

Comment:
Afaik Hyprland capitalized, have to recheck.
A generic fallback to "wlroots" (labwc) backend is needed, it looks to me that atm the taskbar won't work under others than labwc or kwin_wayland.

Reply:
Ran my hyprland setup, panel works. But have to recheck as it looks like env vars have to be in `hyrpland.conf` and not anymore in the start script. But I think we can let hyprland people do necessary things. 

>Hyprthing is for Hyprland what swaymsg is for sway? I do not know much about this.

  yes,` hyprctl clients` would list needed stuff  and there are many commands.

Reply:
@stefonarch Do note that we're lower-casing the env variable when reading it. So this code will work perfectly fine.

Reply:
I saw it working, I didn't notice lowercasing, nice!